src/Xamarin.Android.Build.Tasks/Xamarin.Android.Common.targets

```diff
@@ -2035,6 +2036,19 @@ because xbuild doesn't support framework reference assemblies.
     <Touch Files="$(_AndroidLinkFlag)" AlwaysCreate="true" />
 </Target>
 	
+<Target Name="_GenerateJniMarshalMethods"
+  Condition="'$(JniMarshalMethods)' == 'True' And '$(Configuration)' == 'Release' And '$(AndroidLinkMode)' != 'None'"
```

Comment:
Indeed, `$(JniMarshalMethods)` does appear to control generation as well, wrt https://github.com/xamarin/xamarin-android/pull/1745#discussion_r191732284.

Thus, `$(AndroidGenerateJniMarshalMethods)` seems appropriate.

Reply:
**Code Convention** (also not followed consistently, but let's not make it worse):

For XML, attributes are indented one "tab stop" more than *child elements*.  Additionally, prefer spaces over tabs, and use two-space "tab stops".  Thus:

```xml
<Target Name="_GenerateJniMarshalMethods"
    Condition=" '$(AndroidGenerateJniMarshalMethods)' == 'True' And '$(Configuration)' == 'Release' And '$(AndroidLinkMode)' != 'None' "
    DependsOnTargets="_GetReferenceAssemblyPaths"
    Inputs="@(ResolvedUserAssemblies->'$(MonoAndroidLinkerInputDir)%(Filename)%(Extension)')"
    Outputs="$(_AndroidJniMarshalMethodsFlag)">
  <Exec
      Command="MONO_PATH=$(_XATargetFrameworkDirectories) $(MonoAndroidBinDirectory)\mono $(MonoAndroidBinDirectory)\..\jnimarshalmethod-gen.exe @(ResolvedUserAssemblies->'$(MonoAndroidLinkerInputDir)%(Filename)%(Extension)')"
  />
</Target>
```

Reply:
We *should not* use `'$(Configuration)' == 'Release'`. `$(Configuration)` values are *convention*, and developers are able to add *arbitrary* names. (For example, Java.Interop had a `XAIntegrationRelease` configuration for quite some time.)

*Any* check against `$(Configuration)` is a "code smell." (I think we still have one anyway -- though I can't currently find it -- but, again, we shouldn't make things *worse*.)